src/test/groovy/com/example/CardServiceSpec.groovy

```diff
@@ -0,0 +1,60 @@
+package com.example
+
+import spock.lang.Specification
+
+class CardServiceSpec extends Specification {
+
+    def cardRepository = Stub(ListCardRepository)
+
+    def cardService = new CardService(cardRepository)
+
+    def "should return empty list of cards"() {
+        given:
+        cardRepository.findAll() >> { [] }
+
+        when:
+        def cards = cardService.findAll()
+
+        then:
+        cards.size() == 0
+    }
+
+    def "should return list with many elements"() {
+        given:
+        def firstCard = new Card("foo", Collections.singletonList("baz"))
+        def secondCard = new Card("bar", Collections.singletonList("baz"))
+        cardRepository.findAll() >> { [firstCard, secondCard] }
+
+        when:
+        def cards = cardService.findAll()
+
+        then:
+        cards as List == [secondCard, firstCard] as List
```

Comment:
Zmienić się może jeżeli zmienimy kierunek co wiąże się ze zmianą w kodzie jak i w teście, może zmienić się również rozmiar, ale mamy to zahardcodowane, więc zakładam, że się to nie będzie zmieniać. Zmieni się to jak będzie zapięte Mongo.